tigerpath/majors_and_certificates/scripts/verifier.py

```diff
@@ -316,6 +318,65 @@ def _format_courses_output(courses):
                 output[i][j]["settled"] = course["settled"]
     return output
 
+def _parse_year_code(code, year):
+    """
+    Returns whether `year` falls in the range specified by `code`
+    """
+    if isinstance(code, int):  # explicitly specified year as an int
+        return year == code
+    if not code or code.lower() == "default":  # empty indicates default case
+        return True
+    code = code.replace(" ", "")  # strip it of spaces for processing
+    if code.startswith("<="):
+        return year <= int(code[2:])
+    elif code.startswith(">="):
+        return year >= int(code[2:])
+    elif code.startswith("<"):
+        return year < int(code[1:])
+    elif code.startswith(">"):
+        return year > int(code[1:])
+    elif code.startswith("=="):
+        return year == int(code[2:])
+    elif code.startswith("!="):
+        return year != int(code[2:])
+    elif "-" in code:  # range of years (inclusive), such as "2018-2020"
+        fr, to, *_ = code.split("-")
+        return year >= int(fr) and year <= int(to)
+    else:  # just the year is the same as ==
+        return year == int(code)
+
+def _init_year_switch(req, year):
+    """
+    Checks for any year_switch primitives and selects the right subrequirements.
+
+    Any requirement that contains a year_switch is overridden by the first
+    subrequirement of that year_switch whose year code matches the user's
+    class year.
+    Fields in req are overridden by any explicitly listed fields of the
```

Comment:
So if I understand this correctly, it's just collapsing the fields in the "first subrequirement of the year_switch whose year code matches the user's class year" into the parent requirement, potentially overriding any fields that already exist there?

Reply:
Yeah, that's exactly right!

Do you suggest rewording it to make that clearer?

Reply:
No, I think it's probably fine. Just wanted to check my understanding.